src/Decorators/ArrayDecorator.php

```diff
@@ -45,7 +45,25 @@ public function __construct(AutoPresenter $autoPresenter)
      */
     public function canDecorate($subject)
     {
-        return is_array($subject) || $subject instanceof Collection;
+        return (is_array($subject) || $subject instanceof Collection) && $this->isNotIgnoreClass($subject);
+    }
+
+    /**
+     * Ignore a class from decorate.
+     *
+     * @param $subject
+     *
+     * @return bool
+     */
+    public function isNotIgnoreClass($subject)
+    {
+        foreach (config()->get('laravel-auto-presenter.ignore-class-decorate', []) as $class) {
```

Comment:
We can't accept code that looks like this I'm afraid. The config function doesn't exist, so we can't call it. You must resolve the config in the service provider, and pass it in the constructor.